File: engine/engine-model/src/main/scala/tools/packet/buff/CancelForeignBuff.scala
package tools.packet.buff

import client.MapleBuffStat
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class CancelForeignBuff(private var _characterId: Int, private var _statIncreases: java.util.List[MapleBuffStat]) extends PacketInput {
  def characterId: Int = _characterId

  def statIncreases: java.util.List[MapleBuffStat] = _statIncreases

  override def opcode(): SendOpcode = SendOpcode.CANCEL_FOREIGN_BUFF
}

File: engine/engine-model/src/main/java/tools/packet/buff/CancelForeignBuff.java
package tools.packet.buff;

import java.util.List;

import client.MapleBuffStat;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record CancelForeignBuff(Integer characterId, List<MapleBuffStat> statIncreases) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CANCEL_FOREIGN_BUFF;
   }
}
